main.tf

```diff
@@ -47,13 +41,14 @@ resource "aws_security_group" "default" {
 }
 
 resource "aws_redshift_subnet_group" "default" {
-  name       = var.stack
-  subnet_ids = local.subnet_ids
+  count      = local.create_subnet_group ? 1 : 0
+  name       = "redshift-subnet-group-${var.name}"
```

Comment:
I guess you need to create a `local` that either uses the given `redshift_subnet_group` name or constructs a custom name...